index.js

```diff
@@ -114,23 +114,50 @@ module.exports.iterator = (emitter, event, options) => {
 		multiArgs: false
 	}, options);
 
+	let limit = -1;
+	if (options.limit !== undefined) {
+		if (
+			!(
+				options.limit >= 1 &&
+				(options.limit === Infinity || Number.isInteger(options.limit))
+			)
```

Comment:
This `if` statement looks ugly...

Reply:
That's my doing. It's to make it more readable. Otherwise, it's very hard to see which part is attached to which parens.

Reply:
Okay :)